### PR TITLE
add /listrepos end point

### PR DESCRIPTION
**Pull request by remimeunier**

### CODE REVIEW

**Comment by remimeunier** [comment]
un commentaire

File: lib/dev3/slack_messenger/http_client.ex [hunk 1]
```diff
@@ -32,7 +32,8 @@ defmodule Dev3.SlackMessenger.HTTPClient do
                           review_submitted:        ReviewSubmitted,
                           tagged_in_issue:         TaggedInIssue,
                           tagged_in_issue_comment: TaggedInIssueComment,
-                          rate_limit_exceeded:     RateLimitExceededResponse}
+                          rate_limit_exceeded:     RateLimitExceededResponse,
+                          listrepos_response:      ListreposResponse}
```
Comment (BachirC) [inline lib/dev3/slack_messenger/http_client.ex]: To keep consistency with other slash command messages : `list_repos_response` and `ListReposResponse`

File: lib/dev3/slack_messenger/listrepos_respons.ex [hunk 2]
```diff
@@ -0,0 +1,24 @@
+defmodule Dev3.SlackMessenger.HTTPClient.ListreposResponse do
```
Comment (BachirC) [inline lib/dev3/slack_messenger/listrepos_respons.ex]: Relates to the message above.

File: lib/dev3/slack_messenger/listrepos_respons.ex [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  @moduledoc """
+    Defines the Slack response message attachments for command /listrepos.
```
Comment (BachirC) [inline lib/dev3/slack_messenger/listrepos_respons.ex]: just "message", attachments are part of it.

File: lib/dev3/slack_messenger/listrepos_respons.ex [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  @behaviour Dev3.SlackMessenger.HTTPClient
+
+  def build_message(data) do
+    %{text: "", attachments: build_attachments(data)}
+  end
+
+  defp build_attachments(repos) do
+    [%{
+      title: """
+        The following repos are currently watched.
```
Comment (BachirC) [inline lib/dev3/slack_messenger/listrepos_respons.ex]: "currently being watched" sounds better.

File: lib/dev3/slack_messenger/listrepos_respons.ex [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        To watch more, run `/watchrepos`
+        To stop watching, run `/unwatchrepos`
```
Comment (BachirC) [inline lib/dev3/slack_messenger/listrepos_respons.ex]: 👍

File: lib/dev3/web/controllers/api/slack/slash_commands_controller.ex [hunk 6]
```diff
@@ -10,7 +10,7 @@ defmodule Dev3.Web.API.Slack.SlashCommandsController do
 
   plug :verify_token
   plug :assign_user
-  plug :rate_limit, @rate_limiter_conf when action in [:watch_repos, :unwatch_repos]
+  plug :rate_limit, @rate_limiter_conf when action in [:watch_repos, :unwatch_repos, :list_repos]
```
Comment (BachirC) [inline lib/dev3/web/controllers/api/slack/slash_commands_controller.ex]: 👍

File: lib/dev3/web/controllers/api/slack/slash_commands_controller.ex [hunk 7]
```diff
@@ -41,6 +41,16 @@ defmodule Dev3.Web.API.Slack.SlashCommandsController do
     send_resp(conn, :ok, "")
   end
 
+  @doc """
+    Endpoint for Slack /listrepos command.
+    The command doesn't take any arguments.
+    It shows the list of watch repos.
```
Comment (BachirC) [inline lib/dev3/web/controllers/api/slack/slash_commands_controller.ex]: watched

File: test/web/controllers/api/slack/slash_commands_controller_test.exs [hunk 8]
```diff
@@ -107,6 +107,16 @@ defmodule Dev3.Web.API.Slack.SlashCommandsControllerTest do
     end
   end
 
+  describe "/listrepos" do
```
Comment (BachirC) [inline test/web/controllers/api/slack/slash_commands_controller_test.exs]: 👍

**Review by BachirC — CHANGES_REQUESTED** [review]
Nice work, some minor fixes

File: lib/dev3/slack_messenger/http_client.ex [hunk 9]
```diff
@@ -33,7 +33,8 @@ defmodule Dev3.SlackMessenger.HTTPClient do
                           tagged_in_issue:               TaggedInIssue,
                           tagged_in_issue_comment:       TaggedInIssueComment,
                           rate_limit_exceeded:           RateLimitExceededResponse,
-                          notify_owner_on_issue_comment: NotifyOwnerOnIssueComment}
+                          notify_owner_on_issue_comment: NotifyOwnerOnIssueComment,
+                          listrepos_response:            ListreposResponse}
```
Comment (BachirC) [inline lib/dev3/slack_messenger/http_client.ex]: Still didn't fix `ListReposResponse`. Repos with capital R.

**Review by BachirC — CHANGES_REQUESTED** [review]
Almost :p

**Comment by remimeunier** [comment]
Won't do

**Pull request closed without merging**